pkg/libmount/mount_table.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package libmount
+
+import (
+	"bufio"
+	"os"
+)
+
+const (
+	MNT_FMT_GUESS MountTabFormat = iota
+	MNT_FMT_FSTAB
+	MNT_FMT_MOUNTINFO
+	MNT_FMT_UTAB
+	MNT_FMT_SWAPS
+	MNT_FMT_MTAB = MNT_FMT_FSTAB
+)
+
+type MountTabFormat int
+type MountTabOpt func(*MountTab) error
+
+type MountTab struct {
+	format   MountTabFormat
+	fileName string
+	/* skipped entries
+	int		comms;
+	char		*comm_intro;
+	char		*comm_tail;
+	struct libmnt_cache *cache;
+	void		*fltrcb_data;
+	*/
+	entries      []*Filesystem
+	allowFilters []FsFilter
+	denyFilters  []FsFilter
+	// void		*userdata;
+}
+
+func NewMountTab(opts ...MountTabOpt) (*MountTab, error) {
+	mt := MountTab{}
+	for _, opt := range opts {
+		opt(&mt)
```

Comment:
Do we need to handle the error here?

Reply:
yes, we need to. in the case of the `FromFile` option, an error is returned if the file is not found. have added code to return on error.